src/main/java/de/tum/in/www1/artemis/service/QuizExerciseService.java

```diff
@@ -298,6 +298,24 @@ public QuizExercise findOneWithQuestionsAndStatistics(Long quizExerciseId) {
         return quizExercises;
     }
 
+    /**
+     * Get all quiz exercises for the given exam.
+     *
+     * @param examId the id of the exam
+     * @return the entity
+     */
+    public List<QuizExercise> findByExamId(Long examId) {
+        List<QuizExercise> quizExercises = quizExerciseRepository.findByExamId(examId);
+        User user = userRepository.getUserWithGroupsAndAuthorities();
+        if (quizExercises.size() > 0) {
+            Course course = quizExercises.get(0).getCourseViaExerciseGroupOrCourseMember();
```

Comment:
Can it be possible that `quizExercises` can have exercises belonging to different courses?

Reply:
Well since an exam is held in a single course, it should not be possible that quizExercises from the same exam have different courses